### PR TITLE
Navbar with Logout

### LINKED ISSUE #9: F1.2.2 The service shall allow users to log out of their own account.

### PR DESCRIPTION
**Pull request by wr1159**
Resolves #9

## Changes
* Use app subdirectory eg: app/user-settings/:id
* Add screen responsive navbar
  * Active tab selection
  * Settings routing
  * Logout functionality

There's also a bug where the first login fails which I suspect is because of the Context

## Demo

https://github.com/user-attachments/assets/5d33f3f1-6b94-4e88-8235-045470a18f62

### CODE REVIEW

**Review by SelwynAng — CHANGES_REQUESTED** [review]
Everything looks okay to me. 

As for the bug, why the user fails to log in for the 1st time, I think it has to do with the return value of `login` function in the `auth-context.tsx` file. Currently, that function is returning the `User` state variable instead of the `userData` information directly from the API response. The `User` state variable might not be the most updated due to how state changes are async. So, should try using the `userData` instead.

File: frontend/components/auth/login-form.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ export function LoginForm() {
     event.preventDefault();
     const user = await auth?.login(email, password);
     if (user?.isAdmin) {
-      router.push("/admin-user-management");
+      router.push("/app/admin-user-management");
     } else if (user) {
       router.push("/");
```
[the remark this answers is not in the document]
Comment (wr1159) [reply frontend/components/auth/login-form.tsx:35]: yep, we can start with the question page when it's created

**Review by SelwynAng — APPROVED** [review]
LGTM!

File: frontend/components/admin-user-management/admin-user-management.tsx [hunk 2]
```diff
@@ -123,10 +100,7 @@ export default function AdminUserManagement() {
                 <Button variant="outline" className="mr-2" onClick={() => {}}>
                   Edit
                 </Button>
-                <Button
-                  variant="destructive"
-                  onClick={() => handleDelete(user.id)}
-                >
+                <Button variant="destructive" onClick={() => {}}>
```
Comment (dylkaw) [inline frontend/components/admin-user-management/admin-user-management.tsx:103]: Delete button functionality seems to have regressed here.

**Review by dylkaw — APPROVED** [review]
LGTM!

**Pull request merged**